azurerm/config.go

```diff
@@ -189,6 +189,7 @@ type ArmClient struct {
 	// Databases
 	mariadbDatabasesClient                   mariadb.DatabasesClient
 	mariadbServersClient                     mariadb.ServersClient
+	mariadbFirewallRulesClient               mariadb.FirewallRulesClient
```

Comment:
the reason you are seeing that error is you haven't registered the client. You'll have to add something like:
```
	c.BackendClient = apimanagement.NewBackendClientWithBaseURI(o.ResourceManagerEndpoint, o.SubscriptionId)
	o.ConfigureClient(&c.BackendClient.Client, o.ResourceManagerAuthorizer)
```

in the appropriate area.